domain-server/src/DomainServerSettingsManager.cpp

```diff
@@ -881,24 +893,40 @@ void DomainServerSettingsManager::processNodeKickRequestPacket(QSharedPointer<Re
                 if (!verifiedUsername.isEmpty()) {
                     // if we have a verified user name for this user, we first apply the kick to the username
 
-                    // check if there were already permissions
-                    bool hadPermissions = havePermissionsForName(verifiedUsername);
-
-                    // grab or create permissions for the given username
-                    auto userPermissions = _agentPermissions[matchingNode->getPermissions().getKey()];
-
-                    newPermissions = !hadPermissions || userPermissions->can(NodePermissions::Permission::canConnectToDomain);
-
-                    // ensure that the connect permission is clear
-                    userPermissions->clear(NodePermissions::Permission::canConnectToDomain);
+                    // if we have optional ban parameters, we should ban the username based on the parameter
+                    if (hasOptionalBanParameters) {
+                        if (banByUsername) {
+                            // check if there were already permissions
+                            bool hadPermissions = havePermissionsForName(verifiedUsername);
+        
+                            // grab or create permissions for the given username
+                            auto userPermissions = _agentPermissions[matchingNode->getPermissions().getKey()];
+                            
+                            newPermissions = !hadPermissions || userPermissions->can(NodePermissions::Permission::canConnectToDomain);
+        
+                            // ensure that the connect permission is clear
+                            userPermissions->clear(NodePermissions::Permission::canConnectToDomain);
+                        }
+                    } else {
+                        // check if there were already permissions
+                        bool hadPermissions = havePermissionsForName(verifiedUsername);
+    
+                        // grab or create permissions for the given username
+                        auto userPermissions = _agentPermissions[matchingNode->getPermissions().getKey()];
+                        
+                        newPermissions = !hadPermissions || userPermissions->can(NodePermissions::Permission::canConnectToDomain);
+    
+                        // ensure that the connect permission is clear
+                        userPermissions->clear(NodePermissions::Permission::canConnectToDomain);
+                    }
```

Comment:
Could simplify this to one `if()` statement, to avoid repeating code...
```
if (!hasOptionalBanParameters || banByUserName) {
    // check if there were already permissions
   bool hadPermissions = havePermissionsForName(verifiedUsername);
   
   ...
   
}
```

Reply:
+1